src/eddymotion/model/_sklearn.py

```diff
@@ -98,98 +99,225 @@
+            from sklearn.utils.optimize import _check_optimize_result
+
+            for name in LBFGS_CONFIGURABLE_OPTIONS:
+                if (value := getattr(self, name, None)) is not None:
+                    options[name] = value
+
+            opt_res = optimize.minimize(
+                obj_func,
+                initial_theta,
+                method="L-BFGS-B",
+                bounds=bounds,
+                jac=self.eval_gradient,
+                options=options,
+                args=(self.eval_gradient,),
+                tol=self.tol,
+            )
+            _check_optimize_result("lbfgs", opt_res)
+            return opt_res.x, opt_res.fun
+
+        if isinstance(self.optimizer, str) and self.optimizer in CONFIGURABLE_OPTIONS:
+            for name in CONFIGURABLE_OPTIONS[self.optimizer]:
+                if (value := getattr(self, name, None)) is not None:
+                    options[name] = value
+
+            opt_res = optimize.minimize(
+                obj_func,
+                initial_theta,
+                method=self.optimizer,
+                bounds=bounds,
+                jac=self.eval_gradient,
+                options=options,
+                args=(self.eval_gradient,),
+                tol=self.tol,
+            )
+            return opt_res.x, opt_res.fun
+
+        if callable(self.optimizer):
+            return self.optimizer(obj_func, initial_theta, bounds=bounds)
+
+        raise ValueError(f"Unknown optimizer {self.optimizer}.")
+
+    def fit(self, X, y):
+        r"""Fit Gaussian process regression model.
+
+        This overshadows Scikit-Learn's implementation to remove access over
+        the normalization of y.
+
+        Instead, and following Anderson 2015 and what it looks like from
+        `FSL's souce code <https://git.fmrib.ox.ac.uk/fsl/eddy/-/blob/\
+2480dda293d4cec83014454db3a193b87921f6b0/DiffusionGP.cpp#L218>`__,
+        our fit method should also remove the mean.
+
+        From their paper (p. 167, end of first column):
+
+            Typically one just substracts the mean (:math:`\hat{\mathbf{f}}`)
+            from :math:`\mathbf{f}` and then add it back to
+            :math:`f^{*}`, which is analogous to what is often done in
+            "traditional" regression.
+
+        It is still unclear to me how :math:`f^{*}` is interpolated from
+        :math:`\hat{\mathbf{f}}` (grand mean?), or whether the fitted model
+        accounts for that factor (I doubt it).
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or list of object
+            Feature vectors or other representations of training data.
+
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values.
+
+        Returns
+        -------
+        :obj:`~sklearn.gaussian_process.GaussianProcessRegressor`
+            GaussianProcessRegressor class instance.
+        """
+        super().fit(X, y - y.mean(1)[:, None])
```

Comment:
Related to the comment in https://github.com/nipreps/eddymotion/pull/244#issuecomment-2439882062, 

> (...) calculating the mean along the voxels axis obtaining one mean per b-vector) (...)

and the https://github.com/nipreps/eddymotion/commit/31362568767773b94c780abbe40e5a793ecef43a commit message

> They certainly removed the mean (I don't know if gradient-wise mean or the grand mean),

I'm more inclined to think that the mean should be computed over the b-vecs instead of the voxels, as our `x` are the b-vecs and the data lives on the sphere. But from their `mean_correct` method `m` seems to index each gradient volume, and `s` seems to index each voxel, so it looks like they are averaging over voxels. :pensive:.